nvflare/private/fed/app/simulator/simulator_worker.py

```diff
@@ -0,0 +1,221 @@
+            deploy_args = data[SimulatorConstants.DEPLOY_ARGS]
+
+            client.initialize_fobs()
+
+            app_root = os.path.join(args.workspace, SimulatorConstants.JOB_NAME, "app_" + client.client_name)
+            app_custom_folder = os.path.join(app_root, "custom")
+            sys.path.append(app_custom_folder)
+
+            servers = [{t["name"]: t["service"]} for t in client_config.get("servers")]
+            admin_agent = self.create_admin_agent(sorted(servers)[0], client, deploy_args)
+            admin_agent.start()
+            while True:
+                interval, stop_run = self.do_one_task(client)
+                conn.send(stop_run)
+
+                continue_run = conn.recv()
+                if not continue_run:
+                    self.release_resources(client)
+                    break
+                time.sleep(interval)
+
+        except EOFError:
+            self.logger.info("Client run completed.")
+        except BaseException as error:
+            self.logger.error(error)
+        finally:
+            if admin_agent:
+                admin_agent.shutdown()
+
+
+def _create_connection(listen_port):
+    address = ("localhost", int(listen_port))
+    listener = Listener(address, authkey=CommunicationMetaData.CHILD_PASSWORD.encode())
+    conn = listener.accept()
+    return conn
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--workspace", "-o", type=str, help="WORKSPACE folder", required=True)
+    parser.add_argument("--client", type=str, help="Client name", required=True)
+    parser.add_argument("--port", type=str, help="Listen port", required=True)
+    args = parser.parse_args()
+
+    log_config_file_path = os.path.join(args.workspace, "startup", "log.config")
+    if not os.path.isfile(log_config_file_path):
+        log_config_file_path = os.path.join(os.path.dirname(__file__), "resource/log.config")
+    logging.config.fileConfig(fname=log_config_file_path, disable_existing_loggers=False)
+    log_file = os.path.join(args.workspace, SimulatorConstants.JOB_NAME, "log.txt")
+    add_logfile_handler(log_file)
+
+    workspace = os.path.join(args.workspace, SimulatorConstants.JOB_NAME, "app_" + args.client)
+    os.chdir(workspace)
+    AuthorizationService.initialize(EmptyAuthorizer())
+    AuditService.initialize(audit_file_name=WorkspaceConstants.AUDIT_LOG)
+
+    conn = _create_connection(args.port)
+
+    # start parent process checking thread
+    parent_pid = os.getppid()
+    stop_event = threading.Event()
+    thread = threading.Thread(target=check_parent_alive, args=(parent_pid, stop_event))
+    thread.start()
+
+    try:
+        task_worker = ClientTaskWorker()
+        task_worker.run(args, conn)
+    finally:
+        stop_event.set()
+        conn.close()
+        AuditService.close()
+
+
+if __name__ == "__main__":
+    """
+    This is the main program of simulator worker process when running the NVFlare Simulator using multi-GPU.
```

Comment:
even if single GPU we are using this script to run.

Suggest change the description

```
This is the main program of simulator worker process when running the NVFlare Simulator
```

Reply:
done